Dockerfile.assisted-installer-controller.ocp

```diff
@@ -1,12 +1,12 @@
-FROM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16 AS builder
+FROM registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.20-openshift-4.16 AS builder
 
 WORKDIR /go/src/github.com/openshift/assisted-installer
 ENV GOFLAGS="-mod=vendor"
 
 COPY . .
 RUN make controller
 
-FROM registry.ci.openshift.org/ocp/4.16:base-rhel9
+FROM registry.ci.openshift.org/ocp/4.16:base
```

Comment:
@JoelSpeed does it sound correct this "downgrade" (as well as the other changes in this PR)?